src/Build/BuildCheck/Checks/NoEnvironmentVariablePropertyCheck.cs

```diff
@@ -39,39 +44,53 @@ public override void Initialize(ConfigurationContext configurationContext)
         foreach (CustomConfigurationData customConfigurationData in configurationContext.CustomConfigurationData)
         {
             bool? isVerboseEnvVarOutput = GetVerboseEnvVarOutputConfig(customConfigurationData, RuleId);
-            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;           
+            _isVerboseEnvVarOutput = isVerboseEnvVarOutput.HasValue && isVerboseEnvVarOutput.Value;
         }
+
+        CheckScopeClassifier.NotifyOnScopingReadiness += (string? projectFilePath) =>
+        {
+            while (_buildCheckResults.Count > 0)
+            {
+                BuildCheckResult result = _buildCheckResults.Pop();
+                if (!CheckScopeClassifier.IsActionInObservedScope(_scope, result.Location.File, projectFilePath ?? string.Empty))
```

Comment:
This feels incorrect. The `projectFile` argument should be indicating in which project file build context did the environment variable read occured - so it needs to be drawn from the environment variable read data. Instead here it is drawn from the project that finished evaluation - which can be completely unrelated.

Reply:
The trick is that EnvironmentRead event is aware of the file where this variable is detected (it might be an import), the source project is not visible from this event.